### DISCUSSION TITLE
Properly usage of `oneshot` channel

### DISCUSSION THREAD

**Discussion by BratSinot** [tokio-rs/tokio#4950] [Q&A]
P.S. Extent of this: https://github.com/tokio-rs/tokio/discussions/4776

So, I don't get it what the point of `oneshot` and how to use it right.
I tried to use `oneshot` and sometimes `.await` just hangs forever.

**Comment by Darksonn** [comment]
This kind of question is usually because you aren't actually sending any message on the channel. You should double-check that the message is being sent. For example, did you forget to call the function with the `send`?

However, you are linking to a discussion about cooperative scheduling. That can only be the cause in one situation: You are using `block_on` inside the asynchronous context. Another very similar possibility is that another task is [blocking the thread](https://ryhl.io/blog/async-what-is-blocking/). To investigate those things, I recommend the use of [`tokio-console`](https://github.com/tokio-rs/console). If any tasks shows up as being busy all the time, then that may be the cause.
